tests/Unit/ControlSystem/Systems/Test_Shape.cpp

```diff
@@ -0,0 +1,513 @@
+                               &offsets]() {
+      for (iter.reset(); iter; ++iter) {
+        if (iter.l() == 0 or iter.l() == 1) {
+          continue;
+        }
+        const double freq = freqs[iter()];
+        const double offset = offsets[iter()];
+        if (DerivOrder == 2) {
+          updated_deriv[iter()] =
+              -amplitude * square(freq) * sin(freq * time + offset);
+        } else {
+          updated_deriv[iter()] =
+              -amplitude * cube(freq) * cos(freq * time + offset);
+        }
+      }
+    };
+
+    while (time < final_time) {
+      update_deriv();
+      expected_ah_coefs.update(time, updated_deriv, time + dt);
+      time += dt;
+    }
+
+    // Use func_eps for both because the function is changing so rapidly
+    test_shape_control(generator, make_not_null(&system_helper), initial_time,
+                       final_time, l_max, expected_ah_coefs, func_eps,
+                       func_eps);
+  }
+}
+
+void test_names() {
+  using shape = control_system::Systems::Shape<::ah::ObjectLabel::A, 2>;
+
+  CHECK(pretty_type::name<shape>() == "ShapeA");
+
+  const size_t l_max = 4;
+  SpherepackIterator iter(l_max, l_max);
+  const size_t size = iter.spherepack_array_size();
+
+  // Check known valid indices
+  for (iter.reset(); iter; ++iter) {
+    const auto component_name = shape::component_name(iter(), size);
+    CHECK(component_name.has_value());
+    const std::string check_name =
+        "(l,m)=("s + get_output(iter.l()) + ","s + get_output(iter.m()) + ")"s;
+    CHECK(*component_name == check_name);
+  }
+
+  // Check all indices
+  iter.reset();
+  for (size_t i = 0; i < size; i++) {
+    const auto compact_index = iter.compact_index(i);
+    const auto component_name = shape::component_name(i, size);
+    CHECK(compact_index.has_value() == component_name.has_value());
+    if (component_name.has_value()) {
+      CHECK(*compact_index == iter.current_compact_index());
+      const std::string check_name = "(l,m)=("s + get_output(iter.l()) + ","s +
+                                     get_output(iter.m()) + ")"s;
+      CHECK(*component_name == check_name);
+      ++iter;
+    }
+  }
+}
+
+// [[Timeout, 15]]
+SPECTRE_TEST_CASE("Unit.ControlSystem.Systems.Shape", "[ControlSystem][Unit]") {
+  MAKE_GENERATOR(generator);
+  domain::FunctionsOfTime::register_derived_with_charm();
+
+  test_names();
+
+  // For some of the faster changing AHs, (quadratic and sinusoid) the control
+  // system isn't able to perfectly match the expected map parameters, but it is
+  // able to get the control error to be a small constant offset (rather than
+  // 0). This larger epsilon accounts for the difference in the map parameters
+  // (not their derivatives though which it is able to more accurately match)
+  // based off the constant offset in the control error.
+  const double custom_approx_func_eps = 5.0e-3;
+  // This epsilon is mostly necessary for deriv order 3. For most of deriv order
```

Comment:
Is it expected that a higher-order control system performs significantly worse?

Reply:
These numbers are pretty old. I don't think they are true anymore. I'll remove the confusing comment